File: src/model/Agegroup.java
package model;

public enum Agegroup {
	SCHUETZENKLASSE("Schützenklasse", 0), SENIOREN("Senioren", 1);
	public static int numberOfElements = 2;
	private String name;
	private int id;
	private Agegroup(String name, int id){
		this.name = name;
		this.id = id;
	}
	
	public static Agegroup getAgegroup(String name){
		Agegroup[] agegroups = values();
		for (int i = 0; i < agegroups.length; i++) {
			if(agegroups[i].name.equals(name)){
				return agegroups[i];
			}
		}
		return null;
	}
	
	public String toString(){
		return name;
	}
	
	public int getID(){
		return id;
	}

}

File: Rework/apps/backend/src/domain/ageGroup.ts
// Mirrors model/Agegroup.java - the legacy DB stores these exact German
// display strings in the `agegroup` column, not enum constant names.
export const AGE_GROUPS = ["Schützenklasse", "Senioren"] as const;
export type AgeGroup = (typeof AGE_GROUPS)[number];

export function isAgeGroup(value: string): value is AgeGroup {
  return (AGE_GROUPS as readonly string[]).includes(value);
}
